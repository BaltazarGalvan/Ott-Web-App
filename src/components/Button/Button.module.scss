@use '../../styles/variables';
@use '../../styles/theme';
@use "../../styles/mixins/responsive";

.button {
  position: relative;
  display: inline-flex;
  justify-content: inherit;
  align-items: center;
  min-height: 38px;
  padding: 0 16px;

  color: theme.$btn-default-color;
  background-color: theme.$btn-default-bg;

  font-family: theme.$body-alt-font-family;
  font-size: 16px;
  line-height: 18px;
  text-align: center;
  text-decoration: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.1s ease, transform 0.1s ease;

  > svg {
    height: 18px;
    width: 18px;
  }

  &.fullWidth {
    width: 100%;
  }
  &.outlined {
    border: 1px solid rgba(255, 255, 255, 0.3);
  }

  &.primary,
  &.active:not(.text) {
    color: var(--highlight-contrast-color, white);
    background-color: var(--highlight-color, black);
    &.outlined {
      border-color: var(--highlight-color, white);
    }
  }
  &.text {
    font-size: 18px;
    font-weight: 700;
    opacity: 0.7;
    &:not(:hover) {
      background: none;
    }
    &.active {
      opacity: 1;
    }
  }

  @media (hover: hover) and (pointer: fine) {
    &:hover:not(.active),
    &:focus:not(.active),
    &:hover:is(.text),
    &:focus:is(.text),
    &:active {
      z-index: 1;
      opacity: 1;
      transform: scale(1.1);
      &.fullWidth {
        transform: scale(1.04);
      }
      &.outlined {
        border-color: white;
      }
    }
    &:hover &:focus.active {
      transform: scale(1.1);
      &.fullWidth {
        transform: scale(1.04);
      }
    }
  }
}
.startIcon {
  margin-right: 11px;
  height: 100%;
  display: flex;
  align-items: center;
  > svg {
    height: 20px;
    width: 20px;
    fill: currentColor;
  }
}
