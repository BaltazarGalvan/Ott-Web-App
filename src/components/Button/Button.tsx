import React from 'react';
import classNames from 'classnames';
import { NavLink } from 'react-router-dom';

import styles from './Button.module.scss';

type Color = 'default' | 'primary';

type Variant = 'contained' | 'outlined' | 'text';

type Props = {
  label: string;
  active?: boolean;
  color?: Color;
  fullWidth?: boolean;
  startIcon?: JSX.Element;
  variant?: Variant;
  onClick?: () => void;
  to?: string;
};

const Button: React.FC<Props & React.AriaAttributes> = ({
  label,
  color = 'default',
  startIcon,
  fullWidth = false,
  active = false,
  variant = 'outlined',
  to,
  onClick,
  ...rest
}: Props) => {
  const className = classNames(styles.button, [styles[color]], [styles[variant]], {
    [styles.active]: active,
    [styles.fullWidth]: fullWidth,
  });

  const icon = startIcon ? <div className={styles.startIcon}>{startIcon}</div> : null;
  const span = <span className={styles.buttonLabel}>{label}</span>;

  return to ? (
    <NavLink className={className} to={to} activeClassName={styles.active} {...rest} exact>
      {icon}
      {span}
    </NavLink>
  ) : (
    <button className={className} onClick={onClick} {...rest}>
      {icon}
      {span}
    </button>
  );
};
export default Button;
