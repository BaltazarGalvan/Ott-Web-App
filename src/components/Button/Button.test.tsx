import React from 'react';
import { render } from '@testing-library/react';

import Button from './Button';

describe('<Button>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
});
