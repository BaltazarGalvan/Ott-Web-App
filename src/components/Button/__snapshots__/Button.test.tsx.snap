// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders and matches snapshot 1`] = `
<div>
  <button
    class="button default outlined active"
  >
    <span>
      aa
    </span>
  </button>
</div>
`;
